Java Warnings

@headius / @eregon do either of you mind helping me with these?

```
ext/nio4r/org/nio4r/ByteBuffer.java:27: warning: [serial] serializable class ByteBuffer has no definition of serialVersionUID
public class ByteBuffer extends RubyObject {
       ^
ext/nio4r/org/nio4r/Monitor.java:15: warning: [serial] serializable class Monitor has no definition of serialVersionUID
public class Monitor extends RubyObject {
       ^
ext/nio4r/org/nio4r/Selector.java:206: warning: [rawtypes] found raw type: RubyArray
        RubyArray array = null;
        ^
  missing type arguments for generic class RubyArray<T>
  where T is a type-variable:
    T extends IRubyObject declared in class RubyArray
ext/nio4r/org/nio4r/Selector.java:212: warning: [rawtypes] found raw type: Iterator
        Iterator selectedKeys = this.selector.selectedKeys().iterator();
        ^
  missing type arguments for generic class Iterator<E>
  where E is a type-variable:
    E extends Object declared in interface Iterator
ext/nio4r/org/nio4r/Selector.java:266: warning: [rawtypes] found raw type: Iterator
        Iterator cancelledKeys = this.cancelledKeys.entrySet().iterator();
        ^
  missing type arguments for generic class Iterator<E>
  where E is a type-variable:
    E extends Object declared in interface Iterator
ext/nio4r/org/nio4r/Selector.java:268: warning: [rawtypes] found raw type: Entry
            Map.Entry entry = (Map.Entry)cancelledKeys.next();
               ^
  missing type arguments for generic class Entry<K,V>
  where K,V are type-variables:
    K extends Object declared in interface Entry
    V extends Object declared in interface Entry
ext/nio4r/org/nio4r/Selector.java:26: warning: [serial] serializable class Selector has no definition of serialVersionUID
public class Selector extends RubyObject {
       ^
11 warnings
```

--- a/ext/nio4r/org/nio4r/ByteBuffer.java
+++ b/ext/nio4r/org/nio4r/ByteBuffer.java
@@ -1,6 +1,7 @@
 package org.nio4r;
 
 import java.io.IOException;
+import java.io.Serializable;
 import java.nio.channels.Channel;
 import java.nio.channels.SelectableChannel;
 import java.nio.channels.ReadableByteChannel;
@@ -25,6 +26,7 @@ import org.jruby.runtime.Block;
 created by Upekshej
  */
 public class ByteBuffer extends RubyObject {
+    private static final long serialVersionUID = -6903439483039149324L;
     private java.nio.ByteBuffer byteBuffer;
 
     public static RaiseException newOverflowError(ThreadContext context, String message) {

--- a/ext/nio4r/org/nio4r/Monitor.java
+++ b/ext/nio4r/org/nio4r/Monitor.java
@@ -13,6 +13,7 @@ import org.jruby.runtime.ThreadContext;
 import org.jruby.runtime.builtin.IRubyObject;
 
 public class Monitor extends RubyObject {
+    private static final long serialVersionUID = -3733782997115074794L;
     private SelectionKey key;
     private RubyIO io;
     private IRubyObject interests, selector, value, closed;

--- a/ext/nio4r/org/nio4r/Selector.java
+++ b/ext/nio4r/org/nio4r/Selector.java
@@ -7,7 +7,6 @@ import java.io.IOException;
 import java.nio.channels.Channel;
 import java.nio.channels.SelectableChannel;
 import java.nio.channels.SelectionKey;
-import java.nio.channels.CancelledKeyException;
 
 import org.jruby.Ruby;
 import org.jruby.RubyArray;
@@ -21,9 +20,8 @@ import org.jruby.runtime.ThreadContext;
 import org.jruby.runtime.builtin.IRubyObject;
 import org.jruby.util.io.OpenFile;
 
-import org.nio4r.Monitor;
-
 public class Selector extends RubyObject {
+    private static final long serialVersionUID = -14562818539414873L;
     private java.nio.channels.Selector selector;
     private HashMap<SelectableChannel,SelectionKey> cancelledKeys;
     private volatile boolean wakeupFired;
@@ -203,15 +201,15 @@ public class Selector extends RubyObject {
             return context.nil;
         }
 
-        RubyArray array = null;
+        RubyArray<?> array = null;
 
         if(!block.isGiven()) {
             array = runtime.newArray(this.selector.selectedKeys().size());
         }
 
-        Iterator selectedKeys = this.selector.selectedKeys().iterator();
+        Iterator<SelectionKey> selectedKeys = this.selector.selectedKeys().iterator();
         while(selectedKeys.hasNext()) {
-            SelectionKey key = (SelectionKey)selectedKeys.next();
+            SelectionKey key = selectedKeys.next();
             processKey(key);
 
             selectedKeys.remove();
@@ -263,10 +261,10 @@ public class Selector extends RubyObject {
 
     /* Flush our internal buffer of cancelled keys */
     private void cancelKeys() {
-        Iterator cancelledKeys = this.cancelledKeys.entrySet().iterator();
+        Iterator<Map.Entry<SelectableChannel, SelectionKey>> cancelledKeys = this.cancelledKeys.entrySet().iterator();
         while(cancelledKeys.hasNext()) {
-            Map.Entry entry = (Map.Entry)cancelledKeys.next();
-            SelectionKey key = (SelectionKey)entry.getValue();
+            Map.Entry<SelectableChannel, SelectionKey> entry = cancelledKeys.next();
+            SelectionKey key = entry.getValue();
             key.cancel();
             cancelledKeys.remove();
         }

